### PR TITLE
feat: add nested path resolution for targetOutputKey

### PR DESCRIPTION
**Pull request by Chibionos**
## Summary
- Add `resolve_output_path` utility supporting dot-notation (`summary.status`) and array-indexing (`items[0].name`) for `targetOutputKey` in evaluators
- Wire up path resolution in `LegacyExactMatchEvaluator`, `LegacyJsonSimilarityEvaluator`, `LegacyLlmAsAJudgeEvaluator`, and coded `OutputEvaluator`
- Add `multi-output-agent` sample with 8 evaluator configs covering nested paths, flat keys, array access, and wildcard
- Add e2e testcase (`target-output-key-evals`) auto-discovered by CI integration tests

## Test plan
- [x] 25 unit tests for `resolve_output_path` (wildcard, flat, nested, arrays, mixed, errors, edge cases)
- [x] 12 integration tests for legacy evaluators with nested paths
- [x] All 1956 existing tests pass with no regressions
- [ ] CI integration testcase runs against alpha/cloud/staging environments

### CODE REVIEW

File: src/uipath/eval/evaluators/output_evaluator.py [hunk 1]
```diff
@@ -80,6 +81,8 @@ def _get_full_expected_output(self, evaluation_criteria: T) -> Any:
 
     def _get_expected_output(self, evaluation_criteria: T) -> Any:
         """Load the expected output from the evaluation criteria."""
+        from .._helpers.output_path import resolve_output_path
```
Comment (andrei-rusu) [inline src/uipath/eval/evaluators/output_evaluator.py]: same here
